node/core/approval-voting/src/lib.rs

```diff
@@ -2093,9 +2177,18 @@ async fn launch_approval(
 					"Failed to validate candidate due to internal error",
 				);
 				metrics_guard.take().on_approval_error();
+
```

Comment:
Note that if an error occurs, we do not remove the candidate_hash from the cache. Should the candidate never successfully verify, then it will get evicted since it's priority in the LruCache will only increase if we modify the approval state fo this candidate_hash.

Reply:
I believe this is addressed by the architecture comment above, in which case the `oneshot::Receiver` would return `Err(Canceled)`